src/main/java/de/tum/in/www1/artemis/service/GradingCriterionService.java

```diff
@@ -46,4 +48,39 @@ public GradingCriterion findOne(long gradingCriterionId) {
         return gradingCriterionRepository.findByExerciseIdWithEagerGradingCriteria(exerciseId);
     }
 
+    /**
+     * Calculates the score over all feedback elements that were set using structured grading instructions (SGI)
+     * @param feedback feedback element that was set by SGI
+     * @param totalScore totalScore which is summed up. Starts from 0.0
+     * @param gradingInstructions empty grading instruction Map to collect the used gradingInstructions
+     * @return calculated total score from feedback elements set by SGI
+     */
+    public double computeTotalScore(Feedback feedback, double totalScore, HashMap<Long, Integer> gradingInstructions) {
```

Comment:
If the parameter `totalScore` is always zero, then it doesn't make sense to have it as a parameter. I would suggest to remove it from the signature and provide it as a local variable.

Reply:
No, it is also summed up from the Class where it is called from. Fixed the description